### ISSUE TITLE
no resource to load, early way out angular 16

### ISSUE THREAD

**Issue by Xcaliburx** [VadimDez/ng2-pdf-viewer#1051]
no resource to load, early way out in angular 16. Please help
![image](https://github.com/VadimDez/ng2-pdf-viewer/assets/54830362/334b7bbd-dee7-4220-a31f-050181ae3fc4)

**Comment by teobryer** [comment]
when i add a specific height, it works but i don't really want to do this

**Comment by andrey-semenoff** [comment]
Just updated from 9.1.0 to 10.0.0 and see the same in console.

**Comment by DOrsulak-SOLIEL** [comment]
I'm on Angular 17 and ng-pdf-viewer 10.0.0, hit the same problem. 

Downgrading didnt help me.

When I gave the viewer a height it worked but the scroll bar was then inside the page margin, i needed the viewer to render at full height to get the page to do the scrolling so I tried this and it worked for me:

:host ::ng-deep .ng2-pdf-viewer-container {
  height: auto !important;
  overflow-y: hidden !important;
}

**Comment by vartika-usom** [comment]
I am on angular 16 and ng2-pdf-viewer = 10.0.0. The viewer is not displaying pdf. Same code was working 2 days back.
Any change happened.

**Comment by Hitesh-Prajapati-DFT** [comment]
I am on angular 16.2.6 ng2-pdf-viewer = 10.0.0. The viewer is not displaying pdf. Can anyone help here.

**Comment by Hitesh-Prajapati-DFT** [comment]
@iliasaddab  can you please confirm below code which I added css

<pdf-viewer [src]="someData?.fileURL" (page-rendered)="pageRendered()"
                (after-load-complete)="someEvent($event)" [original-size]="false" [page]="page"
                [render-text]="renderText" [autoresize]="true" [show-all]="false" class="pdf-container"
                style="width: 100%; height: 100%;z-index: 9999;">
            </pdf-viewer>
            
            In my case is not working and getting console page "no resource to load, early way out"

**Comment by iliasaddab** [comment]
> @iliasaddab can you please confirm below code which I added style
> 
> <pdf-viewer [src]="someData?.fileURL" (page-rendered)="pageRendered()" (after-load-complete)="someEvent($event)" [original-size]="false" [page]="page" [render-text]="renderText" [autoresize]="true" [show-all]="false" class="pdf-container" style="width: 100%; height: 100%;z-index: 9999;">
> 
> ```
>         In my case is not working and getting console page "no resource to load, early way out"
> ```

yes try that

**Comment by iliasaddab** [comment]
> @iliasaddab can you please confirm below code which I added style
> 
> <pdf-viewer [src]="someData?.fileURL" (page-rendered)="pageRendered()" (after-load-complete)="someEvent($event)" [original-size]="false" [page]="page" [render-text]="renderText" [autoresize]="true" [show-all]="false" class="pdf-container" style="width: 100%; height: 100%;z-index: 9999;">
> 
> ```
>         In my case is not working and getting console page "no resource to load, early way out"
> ```

oh wait try the parent element with a fixed height and width eg. 1000px
and the pdf component with style="width: 10vh; height: 10vh;z-index: 9999;"
i changed it to VH not %.

**Comment by iliasaddab** [comment]
`div style="width: 1000px; height: 1000px;z-index: 9999;">
<pdf-viewer [src]="someData?.fileURL" (page-rendered)="pageRendered()" (after-load-complete)="someEvent($event)" [original-size]="false" [page]="page" [render-text]="renderText" [autoresize]="true" [show-all]="false" class="pdf-container" style="width: 100vh; height: 10vh;z-index: 9999;">
</div>
`

**Comment by iliasaddab** [comment]
<div style="width: 1000px; height: 1000px;z-index: 9999;">
<pdf-viewer [src]="someData?.fileURL" (page-rendered)="pageRendered()" (after-load-complete)="someEvent($event)" [original-size]="false" [page]="page" [render-text]="renderText" [autoresize]="true" [show-all]="false" class="pdf-container" style="width: 100vh; height: 10vh;z-index: 9999;">
</div>

**Comment by iliasaddab** [comment]
`<div style="width: 1000px; height: 1000px;z-index: 9999;">
<pdf-viewer [src]="someData?.fileURL" (page-rendered)="pageRendered()" (after-load-complete)="someEvent($event)" [original-size]="false" [page]="page" [render-text]="renderText" [autoresize]="true" [show-all]="false" class="pdf-container" style="width: 100vh; height: 100vh;z-index: 9999;">
</div>`

**Comment by iliasaddab** [comment]
`<div style="width: 1000px; height: 1000px;z-index: 9999;">
<pdf-viewer [src]="someData?.fileURL" (page-rendered)="pageRendered()" (after-load-complete)="someEvent($event)" [original-size]="false" [page]="page" [render-text]="renderText" [autoresize]="true" [show-all]="false" class="pdf-container" style="width: 100vh; height: 10vh;z-index: 9999;"></pdf-viewer>
</div>`

**Comment by iliasaddab** [comment]
<div style="width: 1000px; height: 1000px;z-index: 9999;">
<pdf-viewer [src]="someData?.fileURL" (page-rendered)="pageRendered()" (after-load-complete)="someEvent($event)" [original-size]="false" [page]="page" [render-text]="renderText" [autoresize]="true" [show-all]="false" class="pdf-container" style="width: 100vh; height: 100vh;z-index: 9999;"></pdf-viewer>
</div>

**Comment by Hitesh-Prajapati-DFT** [comment]
@iliasaddab  Yes, after adding only height: 100vh, i'm able to view file now thank you.
Can you also explain why we need to add explicitly height. Once again thanks alot.

**Comment by iliasaddab** [comment]
> @iliasaddab  Yes, after adding only height: 100vh, i'm able to view file now thank you.
> Can you also explain why we need to add explicitly height. Once again thanks alot.

Because the page two way binding only works then. See docs.

**Comment by Hitesh-Prajapati-DFT** [comment]
> > @iliasaddab  Yes, after adding only height: 100vh, i'm able to view file now thank you.
> > Can you also explain why we need to add explicitly height. Once again thanks alot.
> 
> Because the page two way binding only works then. See docs.

Ok I found thank you

**Comment by nsanjaneyulu** [comment]
@angular/cli": "17.3.6"
ng2-pdf-viewer": "^10.2.2"
  <div style="width: 1000px; height: 1000px;z-index: 9999; display: block;">
    <pdf-viewer
    [src]="src"
    [render-text]="true"
    style="width: 100vh; height: 100vh;z-index: 9999; display: block;"></pdf-viewer>
  </div>
Console page getting no resource to load, early way out, Please someone help me for that

**Comment by ochstobi** [comment]
I have a similar problem in v 10.2.2. The pdf gets displayed and I get same error "no resource to load, early way out" and the variable holding the Uint8Array of the PDF gets ereased. Its empty then.

**Comment by Waregalias** [comment]
@ochstobi Same issue, i've copied (new ref) the property as a workaround. It's work for me.
